plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2076,6 +2080,63 @@ public void testSplitPruningForFilterOnPartitionColumn()
+            String tableName = table.getName();
+            String sampleValue = testSetup.getSampleValueLiteral();
+            String highValue = testSetup.getHighValueLiteral();
+            // Insert separately to ensure two files with one value each
+            assertUpdate("INSERT INTO " + tableName + " VALUES (1, " + sampleValue + ")", 1);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (2, " + highValue + ")", 1);
+
+            int expectedSplitCount = supportsIcebergFileStatistics(testSetup.getTrinoTypeName()) ? 1 : 2;
+            verifySplitCount("SELECT row_id FROM " + tableName, 2, computeActual("VALUES 1, 2"));
+            verifySplitCount("SELECT row_id FROM " + tableName + " WHERE col = " + sampleValue, expectedSplitCount, computeActual("VALUES 1"));
+            verifySplitCount("SELECT row_id FROM " + tableName + " WHERE col = " + highValue, expectedSplitCount, computeActual("VALUES 2"));
+            verifySplitCount("SELECT row_id FROM " + tableName + " WHERE col > " + sampleValue, expectedSplitCount, computeActual("VALUES 2"));
+            verifySplitCount("SELECT row_id FROM " + tableName + " WHERE col < " + highValue, expectedSplitCount, computeActual("VALUES 1"));
+        }
+    }
+
+    protected abstract boolean supportsIcebergFileStatistics(String typeName);
+
+    @Test(dataProvider = "testDataMappingSmokeTestDataProvider")
+    @Flaky(issue = "https://github.com/trinodb/trino/issues/5172", match = "Couldn't find operator summary, probably due to query statistic collection error")
+    public void testSplitPruningFromDataFileStatistics(DataMappingTestSetup testSetup)
+    {
+        if (testSetup.isUnsupportedType()) {
+            return;
+        }
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_split_pruning_data_file_statistics",
+                // Random double is needed to make sure rows are different. Otherwise compression may deduplicate rows, resulting in only one row group
+                "(col " + testSetup.getTrinoTypeName() + ", r double)")) {
+            String tableName = table.getName();
+            String values =
+                    Stream.concat(
+                            nCopies(100, testSetup.getSampleValueLiteral()).stream(),
+                            nCopies(100, testSetup.getHighValueLiteral()).stream())
+                            .map(value -> "(" + value + ", rand())")
+                            .collect(Collectors.joining(", "));
+            assertUpdate(withSmallRowGroups(), "INSERT INTO " + tableName + " VALUES " + values, 200);
+
+            @Language("SQL") String query = "SELECT * FROM " + tableName + " WHERE col = " + testSetup.getSampleValueLiteral();
+            @Language("SQL") String queryWithoutPredicatePushdown = query + " OR rand() = 42";
```

Comment:
nit: recent intellij versions can color SQL syntax even without per-variable annotation, based on the call below, so `@Language("SQL")` is redundant